src/metabase/search/impl.clj

```diff
@@ -518,6 +522,63 @@
                                    []))))]
     (map annotate search-results)))
 
+(defn- add-collection-effective-location
+  "Batch-hydrates :effective_location and :effective_parent on collection search results. Keeps search results in
+  order."
+  [search-results]
+  (let [collections    (filter #(mi/instance-of? :model/Collection %) search-results)
+        hydrated-colls (t2/hydrate collections :effective_parent)
+        idx->coll      (into {} (map (juxt :id identity) hydrated-colls))]
+    (map (fn [search-result]
+           (if (mi/instance-of? :model/Collection search-result)
+             (idx->coll (:id search-result))
+             (assoc search-result :effective_location nil)))
+         search-results)))
+
+;;; TODO OMG mix of kebab-case and snake_case here going to make me throw up, we should use all kebab-case in Clojure
+;;; land and then convert the stuff that actually gets sent over the wire in the REST API to snake_case in the API
+;;; endpoint itself, not in the search impl.
+(defn serialize
```

Comment:
This function (+ comment, lol) I moved from the scoring namespace since it's not really related to scoring directly and it was hard for me to find where `:collection` gets added to search results

Reply:
Thanks for doing this, I touched this same function and thought "why the heck is this in scoring?" and... forgot to actually do anything about it 😅